test/e2e/tests/ppom-blockaid-approval-farming.spec.js

```diff
@@ -0,0 +1,300 @@
+        },
+      ],
+    })
+    .thenCallback((req) => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: req.body.json.id,
+          result: {
+            from: selectedAddress,
+            gas: '0x19ce5cb',
+            gasUsed: '0xb4c8',
+            input:
+              '0xa22cb465000000000000000000000000b85492afc686d5ca405e3cd4f50b05d358c75ede0000000000000000000000000000000000000000000000000000000000000001',
+            logs: [
+              {
+                address: CONTRACT_ADDRESS.BAYC,
+                data: '0x0000000000000000000000000000000000000000000000000000000000000001',
+                topics: [
+                  '0x17307eab39ab6107e8899845ad3d59bd9653f200f220920489ca2b5937696c31',
+                  `0x000000000000000000000000${selectedAddressWithoutPrefix}`,
+                  '0x000000000000000000000000b85492afc686d5ca405e3cd4f50b05d358c75ede',
+                ],
+              },
+            ],
+            to: CONTRACT_ADDRESS.BAYC,
+            type: 'CALL',
+            value: '0x0',
+          },
+        },
+      };
+    });
+}
+
+describe('PPOM Blockaid Alert - Set Approval to All @no-mmi', function () {
+  it('should show banner alert', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerOnMainnet()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        testSpecificMock: mockInfura,
+        title: this.test.title,
+      },
+
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        const expectedTitle = 'This is a deceptive request';
+        const expectedDescription =
+          'If you approve this request, a third party known for scams might take all your assets.';
+
+        // Click TestDapp button to send JSON-RPC request
+        await driver.clickElement('#maliciousSetApprovalForAll');
+
+        // Wait for confirmation pop-up
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Notification);
+
+        const bannerAlertFoundByTitle = await driver.findElement({
+          css: bannerAlertSelector,
+          text: expectedTitle,
+        });
+        const bannerAlertText = await bannerAlertFoundByTitle.getText();
+
+        assert(
+          bannerAlertFoundByTitle,
+          `Banner alert not found. Expected Title: ${expectedTitle} \nExpected reason: approval_farming\n`,
+        );
+        assert(
+          bannerAlertText.includes(expectedDescription),
+          `Unexpected banner alert description. Expected: ${expectedDescription} \nExpected reason: approval_farming\n`,
+        );
+      },
+    );
+  });
+});
```

Comment:
maybe we can rename the test to ppom-blockaid-setApprovalForAll-farming.spec.js

could be helpful as we do have a check for the approve farming in test/e2e/tests/ppom-blockaid-alert.spec.js

Reply:
Yeah. Updated https://github.com/MetaMask/metamask-extension/pull/21695/commits/295cc265d507001eab9532af019cc266df147054